File: personal-assistant-agent/src/agents/emailAgent.ts
import { BaseAgent } from './base';

export class EmailAgent extends BaseAgent {
    async checkUnreadEmails(maxResults = 10) {
        const { getUnreadEmails } = await import('../tools/GetUnreadEmails');
        return getUnreadEmails(this.google.oauth2Client, maxResults);
    }
}

File: personal-assistant-agent/src/agents/email_agent.py
from typing import Any, List

from src.agents.base import BaseAgent
from src.tools.get_unread_emails import get_unread_emails


class EmailAgent(BaseAgent):
    def check_unread_emails(self, max_results: int = 10) -> List[dict[str, Any]]:
        if not self.google.oauth2_client:
            raise RuntimeError("Google client not authenticated")
        return get_unread_emails(self.google.oauth2_client, max_results)
